doc/source/serve/deploy_model_tutorial.rst

```diff
@@ -0,0 +1,315 @@
+====================================
+End-to-End Model Deployment Tutorial
+====================================
+
+By the end of this tutorial you will have learned how to deploy a machine learning model locally via Ray Serve.
+
+First, install Ray Serve and all of its dependencies by running the following command in your terminal:
+
+.. code-block:: python
+
+  pip install "ray[serve]"
+
+For this tutorial, we'll use `a model <https://huggingface.co/mrm8488/t5-base-finetuned-summarize-news>`_ [#f1] that summarizes news articles.
+Let's take a look at how it works:
+
+.. code-block:: python
+
+  from transformers import AutoTokenizer, AutoModelWithLMHead
+  
+  def summarize(text, max_length=150):
+    tokenizer = AutoTokenizer.from_pretrained("mrm8488/t5-base-finetuned-summarize-news")
+    model = AutoModelWithLMHead.from_pretrained("mrm8488/t5-base-finetuned-summarize-news")
+
+    input_ids = tokenizer.encode(text, return_tensors="pt", add_special_tokens=True)
+    generated_ids = model.generate(input_ids=input_ids, num_beams=2, max_length=max_length,  repetition_penalty=2.5, length_penalty=1.0, early_stopping=True)
+    preds = [tokenizer.decode(g, skip_special_tokens=True, clean_up_tokenization_spaces=True) for g in generated_ids]
+    
+    return preds[0]
```

Comment:
can we just use https://huggingface.co/transformers/main_classes/pipelines.html#transformers.SummarizationPipeline to make this one line?